### ISSUE TITLE
How to start appium for context.startInstrumentation

### ISSUE THREAD

**Issue by herenxue** [appium/appium-uiautomator2-server#498]
for adb shell is am instrument -w io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner
I wanto to start in my android application,how to start use startInstrumentation method, thanks

**Comment by KazuCocoa** [comment]
Do you mean you'd like to start your test app with `adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner` instead of starting your test app process by Appium?

**Comment by herenxue** [comment]
Yes, I want to in my test App to start Appium. Achieve the “adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner” effect 。I find a way is in my app use context.startInstrumentation() method, but I don't know it right or not,. if I want to mplement such functionality in my app, How should I do? thanks.

**Comment by herenxue** [comment]
> 您的意思是您想启动测试应用程序而不是通过 Appium 启动测试应用程序过程吗？`adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner`

Yes, I want to in my test App to start Appium. Achieve the “adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner” effect 。I find a way is in my app use context.startInstrumentation() method, but I don't know it right or not,. if I want to mplement such functionality in my app, How should I do? thanks.

**Comment by herenxue** [comment]
> 

I know you means, but my test app in mobile phone has not permission to exec shell commods. I should user Intent to start or startInstrumentation or others in my test app?

**Comment by herenxue** [comment]
I want implement the function is in my test app, to start the test AppiumUiAutomator2Server（In terminal is input adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner).

**Comment by herenxue** [comment]
You cant imagine if I install the two appium test apk.But in normal I must input adb shell is am instrument -w your.test.app/androidx.test.runner.AndroidJUnitRunner in terminal.But I want my test app auto start the appium.It means I didn't to input adb shell commond.Just launch my test app,and then across my test app to start appium（It is equivalent to we input adb shell commond in terminal）?

**Comment by mykola-mokhnach** [comment]
> I know you means, but my test app in mobile phone has not permission to exec shell commods. I should user Intent to start or startInstrumentation or others in my test app?

It is impossible because of standard Android security restrictions. instrumentation process could only be executed by a shell or a root user. The shell user is only available externally via adb and the latter is only available in rooted devices.

**Comment by herenxue** [comment]
So if I want in my app to start AppiumUiAutomator2Server  is no way?

**Comment by KazuCocoa** [comment]
I see... then perhaps no

**Comment by herenxue** [comment]
Might you consider providing a way that the other app can start the server instead of input adb shell is am instrument -w io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner ? It's can more convenient for use appium test

**Comment by edsion1107** [comment]
https://github.com/RikkaApps/Shizuku
This project should grant you shell privileges and resolve the issue.
